tools/shared/src/main/scala/org/scalajs/core/tools/linker/analyzer/SymbolRequirement.scala

```diff
@@ -0,0 +1,102 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js tools             **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package org.scalajs.core.tools.linker.analyzer
+
+sealed trait SymbolRequirement {
+  final def ++(that: SymbolRequirement): SymbolRequirement =
+    SymbolRequirement.multipleInternal(List(this, that))
+}
+
+object SymbolRequirement {
+  import Nodes._
+
+  def factory(originatingComponent: String): Factory =
+    new Factory(originatingComponent)
+
+  final class Factory private[SymbolRequirement] (origin: String) {
+    def accessModule(moduleName: String): SymbolRequirement =
+      AccessModule(origin, moduleName)
+
+    def callOnModule(moduleName: String, methodName: String): SymbolRequirement =
+      multiple(accessModule(moduleName), callMethod(moduleName, methodName))
+
+    def instantiateClass(className: String,
+        constructor: String): SymbolRequirement = {
+      InstantiateClass(origin, className, constructor)
+    }
+
+    def instantiateClass(className: String,
+        constructors: Traversable[String]): SymbolRequirement = {
```

Comment:
I kind of want to keep these, since it seems to be a common use-case in general:

I have a class/object for runtime support of my linker thingy. I need all its methods.

Probably I should just leave it like it is. If you don't call a ctor of a class, its not instantiated. If you make a difference between constructed and initialized, you are doing it wrong and stuff will break anyways once we inline `init___`.